### DISCUSSION TITLE
upgrade to 0.31.1 from 0.27.1

### DISCUSSION THREAD

**Discussion by hari819** [strimzi/strimzi-kafka-operator#7493] [Q&A]
Hello ,

i am trying to upgrade from upgrade to 0.31.1 from version 0.27.1 , i can see zookeeper and kafka pods getting started fine but cannot see the zookeeper and kafka statefulset definitions in kubernetes ,  is this expected?

@scholzj  , please could you confirm

Thanks,

**Comment by hari819** [comment]
i think the CRDs did not get upgraded , that could be the cause 

is there a way i can do it , i have used helm upgrade --install command

**Comment by hari819** [comment]
is there a way the CRDs can get the changes from 0.31.1 version without deleting the existing ones?

**Comment by scholzj** [comment]
Strimzi 0.31.1 does use the `StrimziPodSets` by default now instead of StatefulSets. If you do `kubectl get strimzipodsets` you should be able to see them. Please check our YouTube channel for more details -> there is a dedicated video and it is also covered in the "What's new videos". 

That said -> the CRDs should be upgraded every upgrade as they change basically in every release.

**Reply by hari819** [reply]
thankyou @scholzj , so i should delete existing CRDs and install new ones ?

please could you suggest if we have some other option , it may be risky in PROD envs

**Reply by hari819** [reply]
i can get the strimzipodsets , i can see ZK and Kafka

**Reply by scholzj** [reply]
>  so i should delete existing CRDs and install new ones ?

No, you should update them only. For example `kubectl apply` or `kubectl replace`.

**Reply by hari819** [reply]
thankyou @scholzj ,  i will run the kubectl apply after the helm upgrade command , before installing the kafka cluster definition,

**Reply by scholzj** [reply]
> installing the kafka cluster definition

But if you are doing upgrade, you should already have the Kafka CR, or?

**Reply by hari819** [reply]
@scholzj  , i am able to update the CRDs, will add a comment to run only during operator upgrades,

thanks again,
